From 979f66afe310e3f0a316d98949c919fd45924056 Mon Sep 17 00:00:00 2001
From: Rusty Bird <rustybird@openmailbox.org>
Date: Wed, 11 Jan 2017 14:58:03 +0000
Subject: [system/sepolicy] Allow setfsuid and setfsgid syscalls for netd

---
 netd.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/netd.te b/netd.te
index 9b44e4b..18aa0a8 100644
--- a/netd.te
+++ b/netd.te
@@ -5,7 +5,7 @@ type netd_exec, exec_type, file_type;
 init_daemon_domain(netd)
 net_domain(netd)
 
-allow netd self:capability { net_admin net_raw kill };
+allow netd self:capability { net_admin net_raw kill setuid setgid };
 # Note: fsetid is deliberately not included above. fsetid checks are
 # triggered by chmod on a directory or file owned by a group other
 # than one of the groups assigned to the current process to see if
-- 
2.1.4

